loadflow/loadflow-validation/src/main/java/com/powsybl/loadflow/validation/ShuntCompensatorsValidation.java

```diff
@@ -0,0 +1,155 @@
+        Objects.requireNonNull(shuntsWriter);
+
+        boolean validated = true;
+        float p = shunt.getTerminal().getP();
+        float q = shunt.getTerminal().getQ();
+        int currentSectionCount = shunt.getCurrentSectionCount();
+        int maximumSectionCount = shunt.getMaximumSectionCount();
+        float bPerSection = shunt.getbPerSection();
+        float qMax = shunt.getProperties().containsKey("qMax") ? Float.parseFloat((String) shunt.getProperties().get("qMax")) : Float.NaN;
+        float nominalV = shunt.getTerminal().getVoltageLevel().getNominalV();
+        Bus bus = shunt.getTerminal().getBusView().getBus();
+        if (bus != null && !Float.isNaN(bus.getV())) {
+            float v = bus.getV();
+            return checkShunts(shunt.getId(), p, q, currentSectionCount, maximumSectionCount, bPerSection, v, qMax, nominalV, config, shuntsWriter);
+        } else if (!Float.isNaN(q) && q != 0) { // if the shunt is disconnected then either “q” is not defined or “q” is 0
+            LOGGER.warn("{} {}: {}: disconnected shunt Q {}", ValidationType.SHUNTS, ValidationUtils.VALIDATION_ERROR, shunt.getId(), q);
+            validated = false;
+        }
+        try {
+            shuntsWriter.write(shunt.getId(), q, Float.NaN, p, currentSectionCount, maximumSectionCount, bPerSection, Float.NaN, shunt.getTerminal().isConnected(), qMax, nominalV, validated);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+        return validated;
+    }
+
+    public static boolean checkShunts(String id, float p, float q, int currentSectionCount, int maximumSectionCount, float bPerSection,
+                                      float v, float qMax, float nominalV, ValidationConfig config, Writer writer) {
+        Objects.requireNonNull(id);
+        Objects.requireNonNull(config);
+        Objects.requireNonNull(writer);
+
+        try (ValidationWriter shuntsWriter = ValidationUtils.createValidationWriter(id, config, writer, ValidationType.SHUNTS)) {
+            return checkShunts(id, p, q, currentSectionCount, maximumSectionCount, bPerSection, v, qMax, nominalV, config, shuntsWriter);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    public static boolean checkShunts(String id, float p, float q, int currentSectionCount, int maximumSectionCount, float bPerSection,
+                                      float v, float qMax, float nominalV, ValidationConfig config, ValidationWriter shuntsWriter) {
+        boolean validated = true;
+        // “p” is always 0 or NaN
+        if (!Float.isNaN(p) && p != 0) {
+            LOGGER.warn("{} {}: {}: P={}", ValidationType.SHUNTS, ValidationUtils.VALIDATION_ERROR, id, p);
+            validated = false;
+        }
+        // 0 lower or equal “currentSectionCount” lower or equal “maximumSectionCount”
+        if (currentSectionCount < 0 || currentSectionCount > maximumSectionCount) {
```

Comment:
I think it’s already checked in the ShuntCompensatorAdder and in the setter of ShuntCompensator